cpp/src/neighbors/ivf_flat/ivf_flat_helpers.cu

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdint>
+
+#include "ivf_flat_helpers.cuh"
+#include <cuvs/neighbors/ivf_flat.hpp>
+
+namespace cuvs::neighbors::ivf_flat::helpers {
+namespace codepacker {
+namespace detail {
+template <typename T, typename IdxT>
+void pack(
```

Comment:
Please move these into "ivf_flat_helpers.cuh" with the other templates so they can be used more broadly than just within this source file if needed.

Reply:
These methods already are in the public API. Should I still move them?

Reply:
Yes, please move the templates to `ivf_flat_helpers.cuh`. I understand they are being instantiated here, but templates are far more useful (and reusable) in header files.